service-workers/service-worker/iframe-controller.https.html

```diff
@@ -0,0 +1,125 @@
+<!DOCTYPE html>
+<title>Service worker inheritance</title>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="resources/test-helpers.sub.js"></script>
+<script>
+
+let active_workers;
+let parent_controller;
+
+promise_test(t => {
+    const script = 'iframe-controller-worker.https.js';
+    const scope1 = 'iframe-controller.https.html';
+    const scope2 = 'resources/blank.html';
+    const script1 = script + '?for_scope1';
+    const script2 = script + '?for_scope2';
+
+    active_workers = service_worker_unregister_and_register(t, script1, scope1)
+      .then(r => {
+          add_completion_callback(_ => r.unregister());
+          return wait_for_state(t, r.installing, 'activated');
+        })
+      .then(_ => {
+          return service_worker_unregister_and_register(t, script2, scope2);
+        })
+      .then(r => {
+          add_completion_callback(_ => r.unregister());
+          return wait_for_state(t, r.installing, 'activated');
+        });
+
+    return active_workers.then(_ => {
+        // This client's active service worker is set to scope1's registration's
+        // active worker by clients.claim() during the activation.
+        parent_controller = navigator.serviceWorker.controller;
+        return with_iframe('')
```

Comment:
Yes, that makes sense. I think there are quite a few tests already that manually remove the iframes. So I made it opt-in to auto remove the iframe by passing options.auto_remove set to true.